frontend/src/App.jsx

```diff
@@ -1,35 +1,35 @@
-import { BrowserRouter, Routes, Route} from 'react-router-dom';
-import FrontPage from './pages/frontPage/FrontPage';
-import CourseDetails from './pages/CourseDetails/CourseDetails';
-import UserDashboard from './pages/UserDashboard/UserDashboard';
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+import FrontPage from "./pages/frontPage/FrontPage";
+import CourseDetails from "./pages/CourseDetails/CourseDetails";
+import UserDashboard from "./pages/UserDashboard/UserDashboard";
 // import Learn from './pages/Learn/Learn';
-import './App.css'
-import { useState } from 'react';
-import Panel from './pages/frontPage/Panel';
+import "./App.css";
+import { useState } from "react";
+import Panel from "./pages/frontPage/Panel";
 import Courses from "./pages/Courses/Courses";
 import TeacherDashboard from "./pages/TeacherDashboard/TeacherDashboard";
 
-
 function App() {
-
   const [isLoggedIn, setIsLoggedIn] = useState(false);
 
   return (
     <BrowserRouter>
       <Routes>
-        <Route path="/" element={<Panel isLoggedIn={isLoggedIn}/>}>
-          <Route path="/frontpage" element={<FrontPage/>} />          
+        <Route path="/" element={<Panel isLoggedIn={isLoggedIn} />}>
+          <Route path="/frontpage" element={<FrontPage />} />
         </Route>
 
-        <Route path="/UserDashboard" element={<UserDashboard/>} />
-        <Route path="/course/:courseName" element={<CourseDetails/>} />
+        <Route path="/UserDashboard/home" element={<UserDashboard />} />
+        <Route path="/UserDashboard/my-courses" element={<UserDashboard />} />
+        <Route path="/UserDashboard/profile" element={<UserDashboard />} />
+        <Route path="/course/:courseName" element={<CourseDetails />} />
         {/* <Route path="/Learn" element={<Learn />} />  */}
 
-         <Route path="/teacher/dashboard" element={<TeacherDashboard />} />
+        <Route path="/teacher/dashboard" element={<TeacherDashboard />} />
```

Comment:
I'm noticing an inconsistency here in our nesting - "/frontpage" is nested under the "/" route, but none of your routes are. And "/UserDashboard/home" is not nested under "/UserDashboard". I'm find with it for now but we should probably make this consistent at some point.